### ISSUE TITLE
What is the capital city gives wrong answer (should be not understood)

### ISSUE THREAD

**Issue by dkatz23238** [leon-ai/leon#122]
### Specs

- Leon version:
- OS (or browser) version: Ubuntu 18.0
- Node.js version: 12.0

### Expected Behavior

Ideally there should be a capital city module :) I would be glad to work on this.

### Actual Behavior

I belive the meaning of life module is being triggered.

The response from lean is:
1 0 1 0 1 0.

### How Do We Reproduce?
 Run a fresh Leon instance and ask "what is the capital of Nigeria?"

**Comment by dkatz23238** [comment]
Please let me know if I should suggest a "capital city" module separately.

**Comment by louistiti** [comment]
Hello @dkatz23238,

Thanks for reporting. Yes I'm aware of that, this is not really a bug but an improvement we could provide.

In the meantime, let's keep this issue opened and see how we can solve it in the future :smiley:

**Comment by dkatz23238** [comment]
Great. How do you think this one should be done? I was thinking of just hardcoding a large dictionary with key values pairs for countries, however this can become difficult to add new features. I can also look into using an API such as the world bank or even wikipedia. What would you suggest?

**Comment by louistiti** [comment]
My preference would be to go with a hardcoded large dictionary to make it work offline and with low-latency behavior.
In most of the cases it is better using a remote server to request an API or whatever, but here the capital cities don't change that often I suppose? Maybe there is a bit of researches to cover to ensure about the general update frequency of capital cities in the world.

What do you think?

**Comment by dkatz23238** [comment]
Agreed, I will look into this frequency and I do agree with the offline option. It would be nice in a future to have leon be able answer a whole host of demographics, geographic, social, and cultural aspects about all countries around the world (Population size, favorite food, ect). Many of these indicators are usually in a similar format in Wikipedia and for these high level metrics I do not think Wikipedia is a bad source. In a future maybe we can create custom parses for Wikipedia to gain these high level metrics, another option is to do world bank or IMF data. These are definitely things I would like to investigate but for now I'll stick to capital cities and probably going to go with the hard coded dictionary. Leaving this long comment as for future reference.

Ill mock up an initial idea for an online mode where the capital city (and potentially other data) can be parsed from Wikipeida as well as the hard coded version for you/community to later decide.

From some initial research it seem only 11 countries have changed their capital cities officially (funny enough including Nigeria).

**Comment by GaboFDC** [comment]
@dkatz23238 are you working on this? if you are busy, I may give it a try

**Comment by dkatz23238** [comment]
@GaboFDC Please do! I have not been able to advance on this project. Thanks!
